### PR TITLE
Create codeql-analysis.yml

### PR DESCRIPTION
**Pull request by brendandburns**

### CODE REVIEW

File: .github/workflows/codeql-analysis.yml [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+# We have attempted to detect the languages in your repository. Please check
+# the `language` matrix defined below to confirm you have the correct set of
+# supported CodeQL languages.
+# ******** NOTE ********
+
+name: "CodeQL"
+
+on:
+  push:
+    branches: [ master, brendandburns-patch-1, readme-update-bburns, test-fix-ghbuild ]
```
Comment (tg123) [inline .github/workflows/codeql-analysis.yml:17]: private branches?

Comment (brendandburns) [reply .github/workflows/codeql-analysis.yml]: removed. (sorry this file was auto-generated)

**Comment by tg123** [comment]
latest `net5` was introduced last PR
as a result, build failed.

seems codeql does not support net5

**Comment by brendandburns** [comment]
Looks like the fix is to add a `global.json` file

https://github.com/github/codeql-action/issues/244

**Pull request merged**